src/sagemaker/workflow/entities.py

```diff
@@ -57,3 +58,68 @@ class Expression(abc.ABC):
     @abc.abstractmethod
     def expr(self) -> RequestType:
         """Get the expression structure for workflow service calls."""
+
+
+class PipelineVariable(Expression):
+    """Base object for pipeline variables
+
+    PipelineVariables must implement the expr property.
+    """
+
+    def __add__(self, other: Union[Expression, PrimitiveType]):
```

Comment:
> Expression -> PipelineVariable
Same concern above

Reply:
> What happens if I write an expression like this 'a' + parameter_instance_count

This is a good call! this will throw `TypeError: can only concatenate str (not "ParameterInteger") to str` as we're not able to change the str's behavior